framework/src/governance/lib/datazone/datazone-gsr-msk-datasource.ts

```diff
@@ -0,0 +1,232 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Duration, Stack } from 'aws-cdk-lib';
+import { CfnProjectMembership } from 'aws-cdk-lib/aws-datazone';
+import { Rule, RuleTargetInput, Schedule } from 'aws-cdk-lib/aws-events';
+import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
+import {
+  Effect,
+  ManagedPolicy,
+  PolicyDocument,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
+import { Function, Runtime, Code } from 'aws-cdk-lib/aws-lambda';
+import { Construct } from 'constructs';
+import { DataZoneGsrMskDataSourceProps } from './datazone-gsr-msk-datasource-props';
+import { TrackedConstruct, TrackedConstructProps } from '../../../utils';
+
+/**
+ * A DataZone custom data source for MSK (Managed Streaming for Kafka) with integration for Glue Schema Registry.
+ *
+ * @example
+ * import { Schedule } from 'aws-cdk-lib/aws-events';
+ *
+ * new dsf.governance.DataZoneGsrMskDataSource(this, 'MskDatasource', {
+ *   domainId: 'aba_dc999t9ime9sss',
+ *   projectId: '999999b3m5cpz',
+ *   registryName: 'MyRegistry',
+ *   clusterName: 'MyCluster',
+ *   runSchedule: Schedule.cron({ minute: '0', hour: '12' }), // Trigger daily at noon
+ *   enableSchemaRegistryEvent: true, // Enable events for Glue Schema Registry changes
+ * });
+ */
+export class DataZoneGsrMskDataSource extends TrackedConstruct {
+
```

Comment:
Can we expose the rest of the resources create for the lambda? We also should consider the bring your own role for this.